libs/hdf-converters/src/ionchannel-mapper.ts

```diff
@@ -0,0 +1,346 @@
+            result.scans.dependency.dependencies =
+              scanSummary.results.data.dependencies;
+          } else {
+            throw new Error('Dependency scan contains no dependencies array');
+          }
+        // We only care about dependencies at the moment
+        default:
+          break;
+      }
+    });
+  } else {
+    throw new Error(
+      `Ion Channel scan_summaries invalid summary data (expecting array, got ${typeof scanSummaries})`
+    );
+  }
+
+  const dependencyGraph: Record<string, ContextualizedDependency> = {};
+
+  // Flatten dependency tree
+  result.scans.dependency.dependencies.forEach((topLevelDependency) => {
+    const flatDependencies = extractAllDependencies(topLevelDependency);
+    flatDependencies.forEach((dependency) => {
+      dependencyGraph[`${dependency.org}/${dependency.name}`] = dependency;
+    });
+  });
+
+  // Associate dependencies with each-other
+  Object.entries(dependencyGraph).forEach(([, dependency]) => {
+    if (Array.isArray(dependency.dependencies)) {
+      dependency.dependencies.forEach((subDependency) => {
+        dependencyGraph[
+          `${subDependency.org}/${subDependency.name}`
+        ].parentDependencies.push(`${dependency.org}/${dependency.name}`);
+      });
+    }
+  });
+
+  Object.entries(dependencyGraph).forEach(([, dependency]) => {
+    result.scans.dependency.contextualizedDependencies.push(dependency);
+  });
+
+  return result;
+}
+
+export class IonChannelAPIMapper {
+  apiKey: string;
+  projectId?: string;
+  teamId?: string;
+  analysisId?: string;
+
+  apiClient: AxiosInstance;
+
+  constructor(
+    apiKey: string,
+    projectId?: string,
+    teamId?: string,
+    analysisId?: string
+  ) {
+    this.apiKey = apiKey;
+    this.projectId = projectId;
+    this.teamId = teamId;
+    this.analysisId = analysisId;
+
+    this.apiClient = axios.create();
+    this.apiClient.defaults.headers.common[
+      'Authorization'
+    ] = `Bearer ${this.apiKey}`;
+    this.apiClient.defaults.headers.common['Accept'] =
+      'application/json, text/plain, */*';
+  }
+
+  async toHdf(): Promise<ExecJSON.Execution> {
+    const analysis = await this.getAnalysis();
+    const mapper = new IonChannelMapper(JSON.stringify(analysis.analysis));
+    return mapper.toHdf();
+  }
+
+  async setTeam(teamName: string) {
+    const availableTeams = await this.getTeams();
+    const foundTeam = availableTeams.find(
+      (team) => team.name.toLowerCase() === teamName.toLowerCase()
+    );
+    if (foundTeam) {
+      this.teamId = foundTeam.id;
+    } else {
+      throw new Error(
+        `Team ${teamName} not found in available teams: ${availableTeams
+          .map((team) => team.name)
+          .join(', ')}`
+      );
+    }
+  }
+
+  async getTeams(): Promise<Team[]> {
+    if (this.apiKey) {
+      return this.apiClient
+        .get('https://api.ionchannel.io/v1/teams/getTeams')
+        .then(({data}) => data.data);
+    } else {
```

Comment:
this happens a few times in this file so i'll only call it out here, but i personally prefer to reduce the indent levels of stuff as much as possible by putting guard clauses at the top and then letting the rest of the code do its thing

in this case i'd do a check if apikey is not defined (in which case throw the error), and then let the rest of the code be at that top level instead of nested in a if statement